### PR TITLE
Add form text input test case

### PR DESCRIPTION
**Pull request by dylan-isaac**
## Summary

Adds a new test case for evaluating accessible form text input patterns. This tests common accessibility requirements for form fields, including labels, helper text, and keyboard interaction.

### Test Coverage

**WCAG Success Criteria:**
-  1.3.1 Info and Relationships
-  2.1.1 Keyboard
-  2.4.6 Headings and Labels
-  3.3.2 Labels or Instructions
-  4.1.2 Name, Role, Value

**Assertions (4 Requirements, 2 Best Practices):**
-  Each text input has an accessible name (R)
-  Each text input has a textbox role (R)
-  Helper text is programmatically associated (R)
-  Text inputs are keyboard focusable (R)
-  Input labels are visible (BP)
-  Required fields are programmatically indicated (BP)

### Examples Included

**Passing:**
-  `pass-native-labels.html` - label[for] with aria-describedby
-  `pass-wrapped-labels.html` - implicit label wrapping
-  `pass-aria-labelledby.html` - aria-labelledby with aria-required

**Failing:**
-  `fail-no-labels.html` - placeholder only (no label)
-  `fail-placeholder-only.html` - styled placeholder anti-pattern
-  `fail-visual-only-labels.html` - span instead of label
-  `fail-helper-not-associated.html` - missing aria-describedby
-  `fail-wrong-role.html` - contenteditable without role="textbox"
-  `fail-not-focusable.html` - tabindex="-1"

### Testing

All 9 examples pass `pytest tests/test_examples.py -k "form-text-input"`.

### CODE REVIEW

**Comment by dylan-isaac** [comment]
@microsoft-github-policy-service agree [company="Enablement Engineering"]

File: test_cases/form-text-input/test.js [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+ * Get all form field wrappers
+ */
+const getFormFields = async (page) => {
+    return await page.locator('.form-field');
+};
+
+module.exports.run = async ({ page, assert, utils }) => {
+
+    // Assertion 1: Each text input has an accessible name (R - WCAG 4.1.2, 1.3.1)
+    await assert("Each text input has an accessible name", async () => {
```
Comment (mfairchild365) [inline test_cases/form-text-input/test.js]: Rather than write our own custom logic to calculate the accessible name, I'd suggest using an existing method, such as the one axe core uses. I /think/ it's here: https://github.com/dequelabs/axe-core/blob/develop/lib/commons/text/accessible-text.js

As a reference, I'm calling a similar axe core method here: https://github.com/microsoft/a11y-llm-eval/blob/main/test_cases/disclosure-widget/test.js#L33

File: test_cases/form-text-input/test.js [hunk 2]
```diff
@@ -0,0 +1,375 @@
...
+                }
+
+                // Also check for text content outside label/input
+                if (!helperElement) {
+                    const allText = el.textContent || '';
+                    const labelText = label ? label.textContent : '';
+                    const inputValue = input.value || '';
+                    const remainingText = allText.replace(labelText, '').replace(inputValue, '').trim();
+
+                    if (remainingText.length > 10) {
```
Comment (mfairchild365) [inline test_cases/form-text-input/test.js]: 10 seems somewhat arbitrary. Some help text may be shorter. I wonder if there is a better way to handle this.

File: test_cases/form-text-input/test.js [hunk 3]
```diff
@@ -0,0 +1,375 @@
...
+        }
+
+        let visibleLabelCount = 0;
+
+        for (let i = 0; i < count; i++) {
+            const input = inputs.nth(i);
+
+            const hasVisibleLabel = await input.evaluate((el) => {
+                // Check for visible <label> element
+                if (el.labels && el.labels.length > 0) {
```
Comment (mfairchild365) [inline test_cases/form-text-input/test.js]: We might be able to leverage axe to determine if the element is visible or not rather than write custom logic.

File: test_cases/form-text-input/test.js [hunk 4]
```diff
@@ -0,0 +1,375 @@
...
+                return false;
+            });
+
+            if (hasVisibleLabel) {
+                visibleLabelCount++;
+            }
+        }
+
+        return visibleLabelCount === count;
+    }, { type: 'BP' });
```
Comment (mfairchild365) [inline test_cases/form-text-input/test.js]: This isn't a BP, it's a normative requirement.

File: test_cases/form-text-input/test.js [hunk 5]
```diff
@@ -0,0 +1,375 @@
...
+        }
+
+        // If fields appear required visually, they should be programmatically indicated
+        // If no visually required fields, we pass (nothing to check)
+        if (visuallyRequiredCount === 0) {
+            return { pass: true, message: "No visually required fields found" };
+        }
+
+        return programmaticallyIndicatedCount === visuallyRequiredCount;
+    }, { type: 'BP' });
```
Comment (mfairchild365) [inline test_cases/form-text-input/test.js]: I like this as a BP :)

File: test_cases/form-text-input/test.js [hunk 6]
```diff
@@ -0,0 +1,375 @@
...
+            if (hasAccessibleName) {
+                passCount++;
+            }
+        }
+
+        return passCount === count;
+    });
+
+    // Assertion 2: Each text input has textbox role (R - WCAG 4.1.2)
+    await assert("Each text input has textbox role", async () => {
```
[the remark this answers is not in the document]
Comment (dylan-isaac) [reply test_cases/form-text-input/test.js]: I restructured the textbox role assertion to avoid circular logic. It now:  
1. Finds .form-field wrappers (structural)  
2. Identifies text input intent by tag/attribute (not role)  
3. Verifies the field contains an element with the proper textbox role  
  
The blocker for fully adopting getByRole('textbox') is the contenteditable case. getByRole('textbox') won't find contenteditable elements that are  
missing role="textbox" - which is exactly what the fail-wrong-role.html test case checks.  
  
I'm questioning whether that's a realistic LLM failure mode for a contact form. Do you think we should:  
1. Keep contenteditable support (covers theoretical edge case)  
2. Simplify to getByRole('textbox') and remove/repurpose the fail-wrong-role test case

File: test_cases/form-text-input/test.js [hunk 7]
```diff
@@ -0,0 +1,375 @@
...
+        }
+
+        let focusableCount = 0;
+
+        for (let i = 0; i < count; i++) {
+            const input = inputs.nth(i);
+
+            const isFocusable = await input.evaluate((el) => {
+                const tabindex = el.getAttribute('tabindex');
+                const isDisabled = el.disabled;
```
[the remark this answers is not in the document]
Comment (dylan-isaac) [reply test_cases/form-text-input/test.js]: Removed the disabled input check entirely per your suggestion. If an input is disabled, it's intentionally not focusable - that's expected behavior, not a failure.

Implementation note: I check keyboard navigability via the tabindex attribute rather than attempting actual focus. I initially tried using input.focus() to test real focusability but discovered that programmatic focus works on elements with tabindex="-1" - they can receive focus via JavaScript but aren't reachable via Tab navigation.

Checking tabindex directly tests what I actually care about: can users Tab to this input?
